tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp

```diff
@@ -463,12 +463,11 @@ void doTransform(
   geometry_msgs::msg::Polygon & poly_out,
   const geometry_msgs::msg::TransformStamped & transform)
 {
-  poly_out.points.clear();
-  for (auto & point : poly_in.points) {
-    geometry_msgs::msg::Point32 point_transformed;
-    doTransform(point, point_transformed, transform);
-    poly_out.points.push_back(point_transformed);
+  std::vector<geometry_msgs::msg::Point32> points_transformed(poly_in.points.size(), {});
+  for (size_t i=0; i < poly_in.points.size(); ++i) {
+    doTransform(poly_in.points[i], points_transformed[i], transform);
   }
+  poly_out.points = points_transformed;
```

Comment:
```suggestion
  poly_out.points.resize(poly_in.points.size());
  for (size_t i = 0; i < poly_in.points.size(); ++i) {
    doTransform(poly_in.points[i], poly_out.points[i], transform);
  }
```
The `tf2::Polygon` cannot handle the `=` assignment cleanly, so the corresponding test fails. But if we want, an easy fix is to just call `reserve` on `poly_out` before we run the doTransform loop, then it will ensure there's space for the referenced point to be moved into the function call. This is beneficial since we're not copying any values.